lib/Sema/CSApply.cpp

```diff
@@ -3432,6 +3432,8 @@ namespace {
               auto &tc = cs.getTypeChecker();
               tc.diagnose(cast->getLoc(), diag::conditional_downcast_foreign,
                           destValueType);
+              tc.diagnose(cast->getLoc(), diag::note_explicitly_compare_cftypeid,
+                          sub->getReferencedDecl().getDecl()->getBaseName().getIdentifier(), destValueType);
```

Comment:
This won't necessarily work for general expressions and will crash without a referenced declaration.  You should check if you have a reference declaration first.  This should also clean up this chain of accesses.

Reply:
You can also put `DeclBaseName` in the diagnostic so you don't need to `getIdentifier()`.